src/Components/Web.JS/src/Boot.Server.Common.ts

```diff
@@ -113,7 +126,7 @@ async function initializeConnection(options: CircuitStartOptions, logger: Logger
 
   const newConnection = connectionBuilder.build();
 
-  newConnection.on('JS.AttachComponent', (componentId, selector) => attachRootComponentToLogicalElement(0, circuit.resolveElement(selector), componentId, false));
```

Comment:
I think this was a bug waiting to happen, and we only hadn't caught it yet because Blazor Server typically starts faster than Blazor WebAssembly. Thankfully this bug is only relevant to .NET 8.

I've changed the "renderer ID" assignment to use statically-assigned IDs to make this mechanism more robust.